src/Xamarin.Android.Build.Tasks/Tests/Directory.Build.targets

```diff
@@ -0,0 +1,35 @@
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <PropertyGroup>
+    <GeneratedSdkVersionFile Condition="'$(GeneratedSdkVersionFile)' ==''">$(BaseIntermediateOutputPath)$(MSBuildProjectName).SdkVersion$(DefaultLanguageSourceExtension)</GeneratedSdkVersionFile>
+    <GenerateSdkVersionInfo Condition="'$(GenerateSdkVersionInfo)' == ''">true</GenerateSdkVersionInfo>
+    <VersionFile>$([MSBuild]::GetDirectoryNameOfFileAbove('$(MSBuildProjectDirectory)', 'ThirdPartyNotices.txt'))\bin\$(Configuration)\Version</VersionFile>
+    <VersionRevFile>$([MSBuild]::GetDirectoryNameOfFileAbove('$(MSBuildProjectDirectory)', 'ThirdPartyNotices.txt'))\bin\$(Configuration)\Version.rev</VersionRevFile>
+    <_SdkVersion Condition="Exists('$(VersionFile)')">$([System.IO.File]::ReadAllText('$(VersionFile)').Trim())</_SdkVersion>
+    <_SdkVersion Condition="Exists('$(VersionRevFile)')">$(_SdkVersion).$([System.IO.File]::ReadAllText('$(VersionRevFile)').Trim())</_SdkVersion>
+  </PropertyGroup>
+    
+  <Target Name="GenerateSdkVersionInfo"
+          Condition="'$(GenerateSdkVersionInfo)' == 'true' and Exists('$(VersionFile)') and ('$(Language)'=='VB' or '$(Language)'=='C#')"
+          BeforeTargets="CoreCompile"
+          DependsOnTargets="PrepareForBuild"
+          Inputs="$(VersionFile);$(VersionRevFile)" 
+          Outputs="$(GeneratedSdkVersionFile)">
+
+    <ItemGroup>
+      <!-- Ensure the generated assemblyinfo file is not already part of the Compile sources, as a workaround for https://github.com/dotnet/sdk/issues/114 -->
+      <Compile Remove="$(GeneratedSdkVersionFile)" />
+      <_SdkVersionInfo Include="System.Reflection.AssemblyMetadata">
+        <_Parameter1>SdkVersion</_Parameter1>
+        <_Parameter2>$(_SdkVersion)</_Parameter2>
+      </_SdkVersionInfo>
+    </ItemGroup>
+
+    <WriteCodeFragment AssemblyAttributes="@(_SdkVersionInfo)" Language="$(Language)" OutputFile="$(GeneratedSdkVersionFile)" >
+      <Output TaskParameter="OutputFile" ItemName="Compile" />
+      <Output TaskParameter="OutputFile" ItemName="FileWrites" />
```

Comment:
If `GenerateSdkVersionInfo` is skipped due to the `Inputs` / `Outputs` being up to date.

Will `$(GeneratedSdkVersionFile)` get added to `@(Compile)` and `@(FileWrites)`?

Should this be at the bottom of the target instead of using `<Output/>`?
```xml
<ItemGroup>
  <Compile Include="$(GeneratedSdkVersionFile)" />
  <FileWrites Include="$(GeneratedSdkVersionFile)" />
</ItemGroup>
```

Reply:
Yes, absolutely!